### DISCUSSION TITLE
Podcasts: Global setting for downloading past episodes

### DISCUSSION THREAD

**Discussion by brownphotographic** [advplyr/audiobookshelf#3725] [Ideas]
As a user I would like to have a global setting (for all podcasts) that allows me to set the number of episodes to download and keep.

Currently this is a per-podcast basis (and doesn't even work ... bug)

Use case: I would like to download a backlog of episodes and just play. This is a common feature in most podcasting apps (since, well the ipod in the early naughties).

**Comment by nichwall** [comment]
Can you provide more context of how setting the episodes to keep is not working?

The default is to keep all podcast episodes and not delete any of them.

**Reply by brownphotographic** [reply]
Sorry that is not what I am saying...

Currently there is no global setting for all podcasts to set the number of episodes to download and keep.

This is the global dialog for all podcasts
![Screenshot from 2024-12-16 10-03-57](https://github.com/user-attachments/assets/c2fe37c3-d895-421e-ae8c-5e25878b86af)

This is the per podcast dialog. Note the settings.
![Screenshot from 2024-12-16 10-06-13](https://github.com/user-attachments/assets/ef456c54-785b-40c5-8b1b-943b1a390fcc)

There should be (and maybe I am missing it) a global setting for all podcasts to allow the user to define how many episodes to download to the library, and how many to keep. All podcasts should inherit that setting. Then allow the user on a per-podcast basis to over-ride the default setting.

**Reply by nichwall** [reply]
Yes, there is not a global setting for this. I'm mostly confused about what you mean by

> Currently this is a per-podcast basis (and doesn't even work ... bug)

**Comment by brownphotographic** [comment]
To provide more context:

I imported all of my podcasts from opml. The default  global setting with no way to change?) that is used to sync podcasts is the date the podcast was imported.

When I check for new episodes, it only looks at those created since today's date. So no podcasts show. I have to then go into every podcast, change the Setting "Look for new episodes after this date ", change the "Limit", then hit "Check and Download New Episodes"


New Feature (please point this out to me if it already exists):
- Global settings for the whole podcast library
- Look for new episodes after date
- Max # of episodes to keep 
- Max # of new episodes to download per check 

![Screenshot from 2024-12-16 10-29-00](https://github.com/user-attachments/assets/daa56113-b86f-4fec-b310-bd680324b9b7)

**Reply by nichwall** [reply]
There is not currently a way to set the download schedule globally or the max number to keep/download.

By default, all podcasts download the most recent 3 episodes that are not already downloaded when checking for episodes, and all episodes are kept.

If you are wanting episodes to download automatically, I would recommend using the download schedule, or the "Find" button on the main podcast screen.

https://www.audiobookshelf.org/guides/podcasts

**Comment by theh0ff** [comment]
+1 on this. Just imported a heap of podcasts and working my way through each one having to download one episode and then set the schedule and number of episodes to keep. please implement a feature where a global podcast setting can be made.
